core/src/main/java/org/apache/calcite/prepare/CalcitePrepareImpl.java

```diff
@@ -590,21 +611,8 @@ <T> CalciteSignature<T> prepare2_(
       Type elementType,
       long maxRowCount,
       CalciteCatalogReader catalogReader,
-      RelOptPlanner planner) {
+      CalcitePreparingStmt preparingStmt) {
```

Comment:
Is CalcitePreparingStmt required? Or could parameter be a superclass?

Reply:
`createSqlValidator` is only available in `CalcitePreparingStmt` so it has to be `CalcitePreparingStmt`

Reply:
I made this public so we can access and assert in `JdbcTest.java`